src/ERC20Permit.sol

```diff
@@ -67,21 +71,28 @@ abstract contract ERC20Permit is IERC20Permit, StatefulERC712 {
         _revertIfInvalidSignature(owner_, _permit(owner_, spender_, value_, deadline_), signature_);
     }
 
+    /// @inheritdoc IERC20
     function transfer(address recipient_, uint256 amount_) external returns (bool success_) {
         _transfer(msg.sender, recipient_, amount_);
         return true;
     }
 
+    /// @inheritdoc IERC20
     function transferFrom(address sender_, address recipient_, uint256 amount_) external returns (bool success_) {
-        _decreaseAllowance(sender_, msg.sender, amount_);
+        uint256 spenderAllowance_ = allowance[sender_][msg.sender]; // Cache `spenderAllowance_` to stack.
```

Comment:
is there any benefit in doing it OZ way - check allowance, revert if not sufficient, and do unchecked block? 

This version looks better, but any gas savings with unchecked and revert 

https://github.com/OpenZeppelin/openzeppelin-contracts/blob/d155600d554d28b583a8ab36dee0849215d48a20/contracts/token/ERC20/ERC20.sol#L307-L313